src/osd/SnapMapper.h

```diff
@@ -330,7 +349,16 @@ class SnapMapper {
   int get_snaps(
     const hobject_t &oid,     ///< [in] oid to get snaps for
     std::set<snapid_t> *snaps ///< [out] snaps
-    ); ///< @return error, -ENOENT if oid is not recorded
+    ) const; ///< @return error, -ENOENT if oid is not recorded
+
+  // alternative interface to the same data:
+  tl::expected<std::set<snapid_t>, mapper_req_t> get_snaps(
+    const hobject_t& hoid) const final;
+
+  /// a get_snaps() that checks for the existence
+  /// of the corresponding mapping ('SNA_') entries
+  tl::expected<std::set<snapid_t>, mapper_req_t> get_verified_snaps(
+    const hobject_t& hoid) const final;
```

Comment:
These new methods need unit tests in test/test_snap_mapper.cc

Reply:
working on that

Reply:
done. Part of this PR